geovin/sd-v3/build/sd-v3/src/index.js

```diff
@@ -0,0 +1,155 @@
+"use strict";
+var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
+    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
+    return new (P || (P = Promise))(function (resolve, reject) {
+        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
+        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
+        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
+        step((generator = generator.apply(thisArg, _arguments || [])).next());
+    });
+};
+var __importDefault = (this && this.__importDefault) || function (mod) {
+    return (mod && mod.__esModule) ? mod : { "default": mod };
+};
+Object.defineProperty(exports, "__esModule", { value: true });
+const SDV = window.SDV;
+const camera_defaults_json_1 = __importDefault(require("../../assets/js/camera-defaults.json"));
+class GeovinSD {
+    constructor(sd_ticket) {
+        this.initial_parameter_values = {};
+        this.sd_ticket = "";
+        this.initialized = false; // it is loaded and ready
+        this.activated = false; // we have loaded the current settings to see
+        this.visible = false; // we can see it
+        this.rendering = false; // it's processing information
+        this.currentSession = null;
+        this.currentViewport = null;
+        this.parameters = null;
+        this.activeScene = "";
+        this.activeWidth = "";
+        this.activeShape = "";
+        //this.init(sd_ticket);
+        this.sd_ticket = sd_ticket;
+        //this.initial_parameter_values = values;
+    }
+    init(values = {}) {
+        return __awaiter(this, void 0, void 0, function* () {
+            this.rendering = true;
+            this.initial_parameter_values = values;
+            $('.sd-curtain span').text('Initializing your model...');
+            $('.loading-bar__inner').css({ width: '33%' });
+            $('.loading-bar__inner').text('33%');
+            yield this.createViewport();
+            yield this.createSession(this.sd_ticket);
+            this.initialized = true;
+            this.availableScenes = this.currentViewport.lightScenes;
+            this.activeScene = this.currentViewport.lightScene;
+            //console.log('light scenes',this.availableScenes,this.activeScene);
+            $('.sd-curtain span').text('Applying selections ...');
+            $('.loading-bar__inner').css({ width: '66%' });
```

Comment:
Anyway to get real time feedback on the percentages here? Or to present the loading state in a different way?

I don't have all the info on how long this typically takes but there could be another approach here. The text feels more helpful because it provides step by step directions.